src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3647,13 +3775,38 @@ ConnectionStateMachine(WorkerSession *session)
 
 			case MULTI_CONNECTION_LOST:
 			{
-				/* managed to connect, but connection was lost */
-				workerPool->activeConnectionCount--;
```

Comment:
Don't we we still need the old behavior in the else case?

Reply:
Actually I now see this was moved to MULTI_CONNECTION_FAILED. Is there another way to reach that state?

Reply:
Hmm, that's a good point, we can directly come to the `MULTI_CONNECTION_FAILED` where we now decrement counter unnecessarily for such cases. Let me see what we can do there to fix that.

Though, practically speaking, Citus aggressively fails the execution of a connection fails, that's probably the reason no tests failed